From 6542328a6412c0e51616db212b1f3ddb74536fa6 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Sun, 21 Sep 2014 01:17:36 +0300
Subject: [PATCH 1/2] ARM: omap3: beagle: Add .dtsi for the LI-5M03 camera
 sensor board

The Leopard Imaging LI-5M03 is an add-on camera sensor board for the
Beagledboard-xM.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Signed-off-by: Alaganraj Sandhanam <alaganraj.sandhanam@gmail.com>
---
 arch/arm/boot/dts/omap3-beagle-xm-li5m03.dtsi | 63 +++++++++++++++++++++++++++
 arch/arm/boot/dts/omap3-beagle-xm.dts         |  2 +
 2 files changed, 65 insertions(+)
 create mode 100644 arch/arm/boot/dts/omap3-beagle-xm-li5m03.dtsi

diff --git a/arch/arm/boot/dts/omap3-beagle-xm-li5m03.dtsi b/arch/arm/boot/dts/omap3-beagle-xm-li5m03.dtsi
new file mode 100644
index 0000000..4e99c69
--- /dev/null
+++ b/arch/arm/boot/dts/omap3-beagle-xm-li5m03.dtsi
@@ -0,0 +1,63 @@
+/*
+ * Device Tree Source for the Beagleboard-xM LI-5M03 add-on camera board
+ *
+ * Copyright (C) 2014 Laurent Pinchart <laurent.pinchart@ideasonboard.com>
+ *
+ * This file is licensed under the terms of the GNU General Public License
+ * version 2.  This program is licensed "as is" without any warranty of any
+ * kind, whether express or implied.
+ */
+
+&i2c2 {
+	clock-frequency = <100000>;
+
+	mt9p031@48 {
+		compatible = "aptina,mt9p031";
+		reg = <0x48>;
+
+		clocks = <&isp 0>;
+		reset-gpios = <&gpio4 2 GPIO_ACTIVE_LOW>;
+
+		vaa-supply = <&hsusb2_power>;
+		vdd-supply = <&vaux3>;
+		vdd_io-supply = <&vaux4>;
+
+		port {
+			mt9p031_out: endpoint {
+				input-clock-frequency = <21000000>;
+				pixel-clock-frequency = <48000000>;
+				remote-endpoint = <&ccdc_ep>;
+			};
+		};
+	};
+};
+
+&isp {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	ports {
+		port@0 {
+			reg = <0>;
+			ccdc_ep: endpoint {
+				remote-endpoint = <&mt9p031_out>;
+				bus-width = <12>;
+				hsync-active = <1>;
+				vsync-active = <1>;
+				pclk-sample = <0>;
+			};
+		};
+	};
+};
+
+&vaux3 {
+	regulator-name = "cam_core";
+	regulator-min-microvolt = <1800000>;
+	regulator-max-microvolt = <1800000>;
+};
+
+&vaux4 {
+	regulator-name = "cam_io";
+	regulator-min-microvolt = <1800000>;
+	regulator-max-microvolt = <1800000>;
+};
diff --git a/arch/arm/boot/dts/omap3-beagle-xm.dts b/arch/arm/boot/dts/omap3-beagle-xm.dts
index b0b8602..c7c58df 100644
--- a/arch/arm/boot/dts/omap3-beagle-xm.dts
+++ b/arch/arm/boot/dts/omap3-beagle-xm.dts
@@ -438,3 +438,5 @@
 		};
 	};
 };
+
+#include "omap3-beagle-xm-li5m03.dtsi"
-- 
1.9.1

